src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2381,8 +2403,12 @@ private CSharpSyntaxNode LexSingleDirective(
 
             directive = _directiveParser.ParseDirective(isActive, endIsActive, afterFirstToken, afterNonWhitespaceOnLine);
 
-            this.AddTrivia(directive, ref triviaList);
-            _directives = directive.ApplyDirectives(_directives);
+            if (!afterNonWhitespaceOnLine)
```

Comment:
what happens to the directive if we are after non-whitespace?  we don't call .AddTrivia below.  is that ok?  can you comment why this is correct in the code?

Reply:
Yes, we don't want the directive in the trivia, we just want to parse it and then discard it and just take the text we parsed and put it into a BadToken which we put into SkippedTokensTrivia - and on *that* we call AddTrivia.

Reply:
please doc.